pytest_rally/plugin.py

```diff
@@ -86,7 +86,8 @@ def current_branch(repo):
         rev = current_branch(repo)
     tfilter = config.getoption("--track-filter")
     if tfilter:
-        tfilter = [t.strip() for t in tfilter.split(",") if t.strip()]
+        # We are currently filtering marks using only the top-level track (e.g. elastic/logs will include every track marked with elastic*)
+        tfilter = [t.strip().split("/")[0] for t in tfilter.split(",") if t.strip()]
```

Comment:
I find it illogical that `--track-filter=elastic/logs` selects all of `elastic`. I think the following should be true:
* `--track-filter=elastic` selects `elastic/*`
* `--track-filter=elastic/logs` selects `elastic/logs`.

Reply:
My understanding is today for `elastic/shared` changes pytest will receive `--track-filter=elastic` and will therefore run all of `elastic/*` tracks. I think that is perfectly OK.